### PR TITLE
feat(blocks): provide a mobile spec patches

### PR DESCRIPTION
**Pull request by L-Sun**
This PR provides a mobile patches for better user experience. Close [BS-1824](https://linear.app/affine-design/issue/BS-1824/移动端去除-slash的响应)

### What Changes
- hide inline reference popup
- use `keyboard-toolbar` instead of slash menu and block toolbar widgets
- a new @ menu for mobile
- update paragraph placeholder description

### How to use?
In the setup stage of editor, add this spec patches to editor
```ts
import {  MobileSpecsPatches } from '@blocksuite/blocks';
import { IS_MOBILE } from '@blocksuite/global/env';

if (IS_MOBILE) {
  editor.pageSpecs.push(MobileSpecsPatches);
  editor.edgelessSpecs.push(MobileSpecsPatches);
}
```

### CODE REVIEW

**Comment by L-Sun** [comment]
* **#8836** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8836?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8836?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/blocks/src/_specs/preset/mobile-patch.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import { AFFINE_FORMAT_BAR_WIDGET } from '../../root-block/widgets/format-bar/format-bar.js';
+import { AFFINE_SLASH_MENU_WIDGET } from '../../root-block/widgets/slash-menu/index.js';
+
+export class MobileSpecsPatches extends LifeCycleWatcher {
+  static override key = 'mobile-patches';
+
+  constructor(std: BlockStdScope) {
+    super(std);
+
+    std.doc.awarenessStore.setFlag('enable_mobile_keyboard_toolbar', true);
```
Comment (Flrande) [inline packages/blocks/src/_specs/preset/mobile-patch.ts:25]: Why we need to set flag here🤔

Comment (L-Sun) [reply packages/blocks/src/_specs/preset/mobile-patch.ts:25]: It may be better to add both widget to root widgets map.

Comment (L-Sun) [reply packages/blocks/src/_specs/preset/mobile-patch.ts:25]: After trying, it is difficult to control component rendering through extension patch. These two `setFlag` lines are a good example:

- `keyboard-toolbar` is an independent widget for the mobile version, and it is easy to control by overriding the `pageRootWidgetViewMap`.
- However, the latter flag is used to control whether to enable the mobile at-menu, and the mobile at-menu shares most of its logic with the desktop version. So, it is difficult to achieve this goal using the spec patch method only.

```ts
// affine-linked-doc-widget

  show = (mode: 'desktop' | 'mobile' = 'desktop') => {
    // ...

    const enableMobile = this.doc.awarenessStore.getFlag(
      'enable_mobile_linked_doc_menu'
    );

    this._show$.value = enableMobile ? mode : 'desktop';
  };

  override render() {
    if (this._show$.value === 'none') return nothing;

    return html`${this._renderInputMask()}
      <blocksuite-portal
        .shadowDom=${false}
        .template=${choose(
          this._show$.value,
          [
            ['desktop', this._renderLinkedDocPopover],
            ['mobile', this._renderLinkedDocMenu],
          ],
          () => html`${nothing}`
        )}
      ></blocksuite-portal>`;
  }
```

In conclusion, I think setting both flags in the constructor of the patch extension is acceptable. They can be removed in the next version release.

**Review by Flrande — APPROVED** [review]
